soc/nxp/imxrt/imxrt118x/soc.c

```diff
@@ -535,6 +547,24 @@ void soc_early_init_hook(void)
 	/* Get trdc and enable all access modes for MBC and MRC of TRDCA and TRDCW */
 	trdc_enable_all_access();
 
+#if  defined(CONFIG_SECOND_CORE_MCUX) && defined(CONFIG_CPU_CORTEX_M33)
+	/**
+	 * Copy CM7 core from flash to memory. Note that depending on where the
+	 * user decided to store CM7 code, this is likely going to read from the
+	 * flexspi while using XIP. Provided we DO NOT WRITE TO THE FLEXSPI,
+	 * this operation is safe.
+	 *
+	 * Note that this copy MUST occur before enabling the M7 caching to
+	 * ensure the data is written directly to RAM (since the M4 core will use it)
+	 */
+	LISTIFY(SEGMENT_NUM, MEMCPY_SEGMENT, (;));
+#endif
+
+// #if DUAL_CORE_MU_ENABLED && CONFIG_CPU_CORTEX_M4
+// 	/* Set boot flag in messaging unit to indicate boot to primary core */
+// 	MU_SetFlags(MU_BASE, BOOT_FLAG);
+// #endif
```

Comment:
dead code needs to be removed

Reply:
Definitely, This is still Work In Progress, I need to solve why CM7 hangs on WFI instruction, then I clean up the core and create proper PR.